src/blspublickey.cpp

```diff
@@ -39,39 +38,25 @@ BLSPublicKey BLSPublicKey::FromG1(const relic::g1_t* pubKey) {
     BLS::AssertInitialized();
     BLSPublicKey pk = BLSPublicKey();
     g1_copy(pk.q, *pubKey);
-    CompressPoint(pk.data, &pk.q);
     return pk;
 }
 
 BLSPublicKey::BLSPublicKey(const BLSPublicKey &pubKey) {
     BLS::AssertInitialized();
-    relic::g1_t tmp;
-    pubKey.GetPoint(tmp);
-    g1_copy(q, tmp);
-    CompressPoint(data, &q);
+    g1_copy(q, pubKey.q);
 }
 
-size_t BLSPublicKey::size() const {
-    return PUBLIC_KEY_SIZE;
+void BLSPublicKey::Serialize(uint8_t *buffer) const {
+    BLS::AssertInitialized();
+    CompressPoint(buffer, &q);
 }
 
-const uint8_t* BLSPublicKey::begin() const {
+std::vector<uint8_t> BLSPublicKey::Serialize() const {
```

Comment:
Why do we need a second serialize method?

Reply:
1. for convenience (otherwise you'd have to do `std::vector<uint8_t> b; b.resize(size); pk.Serialize(b.data())`
2. as a replacement when begin/end/size was needed (which I removed).